flask/app/datasets.py

```diff
@@ -183,6 +183,25 @@ def list_datasets(page: int, limit: int) -> Response:
             .filter(*filters)
         )
 
+    group_code = request.args.get("group_code", type=str)
+    if group_code:
+        subquery = (
+            models.Dataset.query.join(
+                models.groups_datasets_table,
+                models.Dataset.dataset_id
+                == models.groups_datasets_table.columns.dataset_id,
+            )
+            .join(
+                models.Group,
+                models.Group.group_id == models.groups_datasets_table.columns.group_id,
+            )
```

Comment:
I think this can be simplified by joining direction on the relationship like:

```
models.Dataset.query.join(
                models.Dataset.groups
            )
```